### DISCUSSION TITLE
Axis `nice` only appears to affect one end of axis

### DISCUSSION THREAD

**Discussion by knutwannheden** [observablehq/plot#1078] [General]
I created a plot looking like a demography chart and set `nice: true` for the X axis. This only appears to affect one end of the axis (the left one in my case). That seems like a bug to me.

Here is the example code (without data):

```js
Plot.plot({
  x: {
    grid: true,
    label: "← Unbezahlt · Bezahlt →",
    labelAnchor: "center",
    tickFormat: Math.abs,
    nice: true
  },
  marks: [
    Plot.barX(source, Plot.stackX2({
      x: d => d.bezahlt == 'ja' ? d.anzahl : -d.anzahl,
      y: "partei",
      sort: {y: "x", reverse: true},
      fill: "bezahlt",
      title: d => `anzahl: ${d.anzahl}`
    })),
    Plot.text(source, {
      x: d => d.bezahlt == 'ja' ? d.anzahl + 10 : -d.anzahl - 10,
      y: "partei", text: "anzahl",
      // textAnchor: 'end',
    }),
    Plot.ruleX([0])
  ],
  color: {
    type: "categorical",
    domain: ['nein', 'ja'],
  },
  marginLeft: 50,
})
```

And the result: 
<img width="477" alt="image" src="https://user-images.githubusercontent.com/408765/194715993-2cf77a17-7225-4520-b018-61bf447597c6.png">

Note: Regarding another question I posted this example in https://github.com/observablehq/plot/discussions/1073

**Comment by mbostock** [comment]
It affects both the start and end of the scale domain. You can test this by inspecting the scale domain directly (using plot.scale). Note that you can also pass a number to the nice option to control how much it nices. If you can share a live example, I can investigate further but it’s hard to tell from a screenshot.
